Mathlib/Data/Fin/Basic.lean

```diff
@@ -2543,4 +2544,15 @@ protected theorem zero_mul [NeZero n] (k : Fin n) : (0 : Fin n) * k = 0 := by
 
 end Mul
 
+open Qq in
+instance toExpr (n : ℕ) : Lean.ToExpr (Fin n) where
+  toTypeExpr := q(Fin $n)
+  toExpr := match n with
+    | 0 => finZeroElim
+    | k + 1 => fun i => show Q(Fin $n) from
+      have i : Q(Nat) := Lean.mkNatLit i -- raw literal to avoid ofNat-double-wrapping
+      have : Q(NeZero $n) := have : $n =Q $k + 1 := ⟨⟩; by exact q(NeZero.succ)
+      q(OfNat.ofNat $i)
```

Comment:
Don't you need the `Fin $(k+1)` annotation here? How does q know which OfNat instance to use otherwise?

Reply:
There's a `Q(Fin $n)` annotation above.

Reply:
Ah, I missed that!